File: client/src/au/com/codeka/NumberFormatter.java
package au.com.codeka;

import java.text.DecimalFormat;
import java.text.NumberFormat;

/** Helper for formatting numbers in a nice way (with commas, essentially). */
public class NumberFormatter {
    private static NumberFormat format;

    static {
        format = NumberFormat.getInstance();
        ((DecimalFormat) format).applyPattern("#,##0;-#,##0");
    }

    public static String format(int number) {
        return format.format(number);
    }

    public static String format(long number) {
        return format.format(number);
    }

    public static String format(double number) {
        return format.format(Math.ceil(number));
    }

    public static String format(float number) {
        return format.format(Math.ceil(number));
    }
}

File: client/src/main/java/au/com/codeka/warworlds/client/util/NumberFormatter.kt
package au.com.codeka.warworlds.client.util

import java.text.DecimalFormat
import java.text.NumberFormat

/** Helper for formatting numbers in a nice way (with commas, essentially).  */
object NumberFormatter {
  private var format: NumberFormat? = null
  fun format(number: Int): String {
    return format!!.format(number.toLong())
  }

  fun format(number: Long): String {
    return format!!.format(number)
  }

  fun format(number: Double): String {
    return format!!.format(Math.ceil(number))
  }

  @JvmStatic
  fun format(number: Float): String {
    return format!!.format(Math.ceil(number.toDouble()))
  }

  init {
    format = NumberFormat.getInstance()
    (format as DecimalFormat?)!!.applyPattern("#,##0;-#,##0")
  }
}